spec/controllers/submissions_controller_spec.rb

```diff
@@ -0,0 +1,642 @@
+          submission.save
+        end
+      end
+
+      it 'should get an error if it is before assignment due date' do
+        allow(Assignment).to receive(:find) {@assignment}
+        @assignment.expects(:short_identifier).once.returns('a1')
+        @assignment.submission_rule.expects(
+            :can_collect_now?).once.returns(false)
+        get_as @admin,
+               :collect_all_submissions,
+               assignment_id: 1
+        expect(flash[:error]).to eq(I18n.t(
+            'collect_submissions.could_not_collect',
+            assignment_identifier: 'a1'))
+        is_expected.to respond_with(:redirect)
+      end
+
+      it 'should succeed if it is after assignment due date' do
+        @submission_collector = SubmissionCollector.create
+        allow(Assignment).to receive(:find) {@assignment}
+        SubmissionCollector.expects(:instance).returns(@submission_collector)
+        @assignment.expects(:short_identifier).once.returns('a1')
+        @assignment.submission_rule.expects(
+            :can_collect_now?).once.returns(true)
+        @submission_collector.expects(:push_groupings_to_queue).once
+        get_as @admin, :collect_all_submissions, assignment_id: 1, id: 1
+        expect(flash[:success]).to eq(I18n.t(
+            'collect_submissions.collection_job_started',
+            assignment_identifier: 'a1'))
+        is_expected.to respond_with(:redirect)
+      end
+
+      it 'should be able to release submissions' do
+        allow(Assignment).to receive(:find) {@assignment}
+        post_as @admin,
+                :update_submissions,
+                assignment_id: 1,
+                groupings: ([] << @assignment.groupings).flatten,
+                release_results: 'true'
+        is_expected.to respond_with(:success)
+      end
+    end  
+
+    it 'download all files uploaded in a Zip file' do
+      @file1_name = 'TestFile.java'
+      @file2_name = 'SecondFile.go'
+      @file1_content = "Some contents for TestFile.java\n"
+      @file2_content = "Some contents for SecondFile.go\n"
+
+      @group.access_repo do |repo|
+        txn = repo.get_transaction('test')
+        path = File.join(@assignment.repository_folder, @file1_name)
+        txn.add(path, @file1_content, '')
+        path = File.join(@assignment.repository_folder, @file2_name)
+        txn.add(path, @file2_content, '')
+        repo.commit(txn)
+
+        # Generate submission
+        @submission = Submission.generate_new_submission(
+                          @grouping, repo.get_latest_revision)
+      end
+      get_as @admin,
+             :downloads,
+             assignment_id: @assignment.id,
+             id: @submission.id,
+             grouping_id: @grouping.id
+
+      expect('application/zip').to eq(response.header['Content-Type'])
+      is_expected.to respond_with(:success)
+      zip_path = "tmp/#{@assignment.short_identifier}_" +
+          "#{@grouping.group.group_name}_r#{@grouping.group.repo.
```

Comment:
Place the . on the next line, together with the method name.<br>Align the operands of an expression in an assignment spanning multiple lines.